pkg/cmd/server/api/types.go

```diff
@@ -491,6 +491,14 @@ type HTPasswdPasswordIdentityProvider struct {
 
 type LDAPPasswordIdentityProvider struct {
 	api.TypeMeta
+	// ClientConfig is the configuration for the LDAP client connection
+	LDAPClientConfig
+	// Attributes maps LDAP attributes to identities
+	Attributes LDAPAttributeMapping
+}
+
+// LDAPClientConfig holds information for connecting to an LDAP server
+type LDAPClientConfig struct {
 	// URL is an RFC 2255 URL which specifies the LDAP search parameters to use. The syntax of the URL is
 	//    ldap://host:port/basedn?attribute?scope?filter
 	URL string
```

Comment:
The impetus for this refactor is that I need *almost* all of the information inside of the object you created. As it stands before the refactor, you require an ID inside of the Attributes of your object and I do not. It is also difficult to explain to a user why they have to provide an ID mapping for LDAP sync when we don't use that information in the sync job. Unless we have two parallel objects that only differ in how much we want from this field, some sort of refactor I think is required. I understand that making this an opaque string makes the user input easier, so do you think we should just keep two opaque strings and duplicate the rest of the code around them?